repo2docker/buildpacks/guix/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""BuildPack for guix environments"""
+import os
+
+from ..base import BuildPack, BaseImage
+
+
+class GuixBuildPack(BaseImage):
+    """A Guix Package Manager BuildPack"""
+
+    def get_build_env(self):
+        """"""
+        env = super().get_build_env() + [
+            ("GUIX_DIR", "${APP_BASE}/guix")
+        ]
+        return env
+
+    def get_path(self):
+        """Return paths to be added to PATH environment variable"""
+        path = super().get_path()
+        path.insert(0, "${GUIX_DIR}/bin")
+        return path
+
+
+
+    def get_build_scripts(self):
+        """
+        Install a pinned version of Guix package manager,
+        precised in guix-install.bash.
+        """
+        return super().get_build_scripts() + [
+            (
+                "root",
+                """
+                bash /tmp/.local/bin/guix-install.bash
+                """,
+            ),
+
+        ]
+
+    def get_build_script_files(self):
+
+        """Copying guix installation script on the image"""
+        return {
+            "guix/guix-install.bash":
+                "/tmp/.local/bin/guix-install.bash",
+        }
+
+    def get_assemble_scripts(self):
+        """
+        Launch Guix daemon with --disable-chroot to avoid the need
+        of root privileges for the user.
+        Make sure we never use Debian's python by error by renaming it
+        then, as an user install packages listed in manifest.scm,
```

Comment:
@hulecomte What did you mean at this line?

```suggestion
Make sure we never use Debian's python by error by renaming it
then, as an user installs packages listed in manifest.scm,
```
Maybe, state it again and I can help with clarifying it.